### ISSUE TITLE
overlayWith ignores other options if using `options.raw` or `options.create`

### ISSUE THREAD

**Issue by obmelvin** [lovell/sharp#1431]
If you provide a plain object as the first argument, instead of a buffer or string, that contains a `raw` or `create` property the other options provided will have no effect. In this specific example (below) that means the overlaid image will be positioned by the default `gravity: centre` instead of `top` and `left`.

This is due to how the arguments are handled. If you look at `overlayWith` in `lib/composite.js` both `overlay` and `options` are passed to `_createInputDescriptor`, which correctly recognizes that input is a plain object and takes the appropriate action. However,  after this function finishes execution and we reach the `if (is.object(options))` conditional on [line 54](https://github.com/lovell/sharp/blob/master/lib/composite.js#L54) `options` is undefined and thus no other options are processed.

```
const x = 0;
const y = 0;
const width = 100;
const height = 100;
const original = await sharp('img.png');
const redacted = await original.overlayWith({
        top: x,
        left: y,
        create: {
          width,
          height,
          channels: 4,
          background: {
            r: 0,
            g: 0,
            b: 0,
            alpha: 1,
          },
        },
});
await redacted.toFile('redacted.png');
```

I'm happy to fix this myself after verifying that this isn't the desired behavior.

**Comment by lovell** [comment]
Hello, yes this looks like a bug, happy to accept a PR with the fix and test(s) that would have caught this. Thanks for reporting!

**Comment by lovell** [comment]
This should be fixed by the work for #728 - `overlayWith` is being deprecated in favour of a new multi-image `composite` operation.
